Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/jobs/HttpHandlerUtils.java

```diff
@@ -21,30 +21,30 @@
  */
 package com.microsoft.azure.hdinsight.spark.jobs;
 
-import com.microsoft.azure.hdinsight.common.HttpFutureCallback;
-import com.microsoft.azure.hdinsight.common.MultiHttpFutureCallback;
-import com.microsoft.azure.hdinsight.common.task.*;
+import com.microsoft.azure.hdinsight.common.JobViewManager;
 import com.microsoft.azure.hdinsight.sdk.cluster.IClusterDetail;
-import com.microsoft.azure.hdinsight.spark.jobs.framework.IRequest;
-import com.microsoft.azure.hdinsight.spark.jobs.framework.JobRequestDetails;
-import com.microsoft.azure.hdinsight.spark.jobs.framework.RequestDetail;
-import com.microsoft.azure.hdinsight.spark.jobs.framework.HttpRequestType;
 import com.microsoft.azuretools.azurecommons.helpers.NotNull;
-import com.sun.net.httpserver.Headers;
 import com.sun.net.httpserver.HttpExchange;
-import com.sun.net.httpserver.HttpHandler;
 
-import java.io.IOException;
-import java.util.List;
+import java.net.URI;
+import java.util.LinkedHashMap;
+import java.util.Map;
 
-public class JobHttpHandler implements HttpHandler {
-
-    @Override
-    public void handle(HttpExchange httpExchange) throws IOException {
-        JobRequestDetails requestDetail = JobRequestDetails.getJobRequestDetail(httpExchange);
-        // TODO : get message from spark history server
+public class HttpHandlerUtils {
+    public static Map<String, String> splitQuery(URI uri) {
+        Map<String, String> query_pairs = new LinkedHashMap<String, String>();
+        String query = uri.getQuery();
+        String[] pairs = query.split("&");
+        for (String pair : pairs) {
+            int idx = pair.indexOf("=");
+            query_pairs.put(pair.substring(0, idx), pair.substring(idx + 1));
+        }
+        return query_pairs;
```

Comment:
Have you tested if the char `&` or `=` locates in the query pairs value? such as: `name="m&m"&condition="color=read"`